metadataserver/iam.go

```diff
@@ -91,6 +101,13 @@ func newIamProxy(ctx context.Context, router *mux.Router, config types.MetadataS
 	}
 	proxy.roleName = splitRole[1]
 
+	// No need to block here
+	go proxy.startRoleAssumer()
```

Comment:
Do we even need `startRoleAssumer`? If it's always started, why not have:
```
log.Info("Starting role assumer")
		proxy.doAssumeRole(defaultSessionLifetime)
		log.Info("Ran first assume role")

		go proxy.roleAssumer()
```
here?

Reply:
I didn't write this piece of code - I only moved it around.  No idea why.

Reply:
originally, every time we got a request that needed an assumerole, we would call the `startRoleAssumer` function to start the assume role in the background. You should be able to hoist this code to here. That should give you two side effects:

1. Simplicity
2. The IMDS will block on its first attempt to assume role.

Reply:
Sure, that's fine.  PTAL.